src/VenueModel.cpp

```diff
@@ -258,7 +259,13 @@ QVariant VenueModel::data(const QModelIndex &index, int role) const
         }
 
         auto const openingMinutes = openingMinutesVar.toList();
-        return isInRange(openingMinutes[m_currendDayIndex].toMap(), m_currentMinute);
+
+        switch(role) {
+        case VenueModelRoles::Open:
+            return isInRange(openingMinutes[m_currendDayIndex].toMap(), m_currentMinute);
+        case VenueModelRoles::ClosesSoon:
+            return isInRange(openingMinutes[m_currendDayIndex].toMap(), m_currentMinute + MINUTES_CLOSES_SOON);
```

Comment:
Isn't that the wrong logic here? You check whether the restaurant is still open in ```MINUTES_CLOSES_SOON``` minutes. Isn't what we actually would like check whether it is open now, but it will be closed in ```MINUTES_CLOSES_SOON``` minutes?

Reply:
You are right. I addressed the issue and also put a comment in the code.